ethereum/datastructures/src/main/java/tech/pegasys/artemis/datastructures/util/BeaconStateUtil.java

```diff
@@ -176,9 +176,16 @@ static void process_deposit_without_checking_merkle_proof(
   }
 
   public static boolean is_valid_genesis_state(BeaconState state) {
-    return !(state.getGenesis_time().compareTo(MIN_GENESIS_TIME) < 0)
-        && !(get_active_validator_indices(state, UnsignedLong.valueOf(GENESIS_EPOCH)).size()
-            < MIN_GENESIS_ACTIVE_VALIDATOR_COUNT);
+    return isItMinGenesisTimeYet(state) && isThereEnoughNumberOfValidators(state);
+  }
+
+  public static boolean isThereEnoughNumberOfValidators(BeaconState state) {
+    return !(get_active_validator_indices(state, UnsignedLong.valueOf(GENESIS_EPOCH)).size()
+        < MIN_GENESIS_ACTIVE_VALIDATOR_COUNT);
+  }
+
+  public static boolean isItMinGenesisTimeYet(BeaconState state) {
+    return !(state.getGenesis_time().compareTo(MIN_GENESIS_TIME) < 0);
```

Comment:
This is probably personal preference but I'd write this as:
```suggestion
    return state.getGenesis_time().compareTo(MIN_GENESIS_TIME) >= 0;
```

Reply:
yeah me too, idk why it was previously written that way.